src/openforms/forms/api/serializers/logic/action_serializers.py

```diff
@@ -189,6 +192,20 @@ def validate(self, data: dict) -> dict:
                 code="blank",
             )
 
+        # validate format of value for date variable
+        if action_type and action_type == LogicActionTypes.variable:
+            # get related form variable from context of parent serializer
+            form_var = self.parent.parent.context["form_variables"].variables[variable]
+            action_value = data.get("action", {}).get("value")
+
+            if form_var.data_type == FormVariableDataTypes.date:
+                if _ := format_date_value(action_value) == "":
+                    raise serializers.ValidationError(
```

Comment:
I know Silvia said otherwise, but I'm not a fan of this flow control - much rather see a `ValueError` being caught and then `raise ValidationError(...) from exc` for full context.

`format_date_value` is written with the intent that a valid date structure was passed in, this is abusing it outside of its intended usage.